modin/experimental/xgboost/test/test_xgboost.py

```diff
@@ -0,0 +1,301 @@
+)
+@pytest.mark.parametrize(
+    "num_actors",
+    [1, num_cpus, None, modin.config.NPartitions.get() + 1],
+)
+@pytest.mark.parametrize(
+    "data",
+    [
+        (
+            load_breast_cancer(),
+            {"objective": "binary:logistic", "eval_metric": ["logloss", "error"]},
+        ),
+    ],
+    ids=[
+        "load_breast_cancer",
+    ],
+)
+def test_xgb_with_binary_classification_datasets(data, num_actors, modin_type_y):
+    dataset, param = data
+    num_round = 10
+
+    X = dataset.data
+    y = dataset.target
+    xgb_dmatrix = xgboost.DMatrix(X, label=y)
+
+    modin_X = pd.DataFrame(X)
+    modin_y = modin_type_y(y)
+    mxgb_dmatrix = xgb.DMatrix(modin_X, label=modin_y)
+
+    evals_result_xgb = {}
+    evals_result_mxgb = {}
+    verbose_eval = False
+    bst = xgboost.train(
+        param,
+        xgb_dmatrix,
+        num_round,
+        evals_result=evals_result_xgb,
+        evals=[(xgb_dmatrix, "train")],
+        verbose_eval=verbose_eval,
+    )
+    modin_bst = xgb.train(
+        param,
+        mxgb_dmatrix,
+        num_round,
+        evals_result=evals_result_mxgb,
+        evals=[(mxgb_dmatrix, "train")],
+        num_actors=num_actors,
+        verbose_eval=verbose_eval,
+    )
+
+    for par in param["eval_metric"]:
+        assert len(evals_result_xgb["train"][par]) == len(
+            evals_result_xgb["train"][par]
+        )
+        for i in range(len(evals_result_xgb["train"][par])):
+            np.testing.assert_allclose(
+                evals_result_xgb["train"][par][i],
+                evals_result_mxgb["train"][par][i],
+                atol=0.011,
+            )
+
+    predictions = bst.predict(xgb_dmatrix)
+    modin_predictions = modin_bst.predict(mxgb_dmatrix, num_actors=num_actors)
+
+    preds = pd.DataFrame(predictions).apply(lambda x: round(x))
+    modin_preds = modin_predictions.apply(lambda x: round(x))
+
+    val = accuracy_score(y, preds)
+    modin_val = accuracy_score(modin_y, modin_preds)
+
+    np.testing.assert_allclose(val, modin_val, atol=0.002, rtol=0.002)
+
+
+@pytest.mark.parametrize(
+    "modin_type_y",
+    [pd.DataFrame, pd.Series],
+)
+@pytest.mark.parametrize(
+    "num_actors",
+    [1, num_cpus, None, modin.config.NPartitions.get() + 1],
+)
+@pytest.mark.parametrize(
+    "data",
+    [
+        (
+            load_iris(),
+            {
+                "num_class": 3,
+            },
+        ),
+        (
+            load_digits(),
+            {
+                "num_class": 10,
+            },
+        ),
+        (
+            load_wine(),
+            {
+                "num_class": 3,
+            },
+        ),
+    ],
+    ids=[
+        "load_iris",
+        "load_digits",
+        "load_wine",
+    ],
+)
+def test_xgb_with_multiclass_classification_datasets(data, num_actors, modin_type_y):
+    dataset, param_ = data
+    num_round = 10
+    part_param = {"objective": "multi:softprob", "eval_metric": "mlogloss"}
```

Comment:
This should probably go to `data` parameter,

Reply:
Can we go the same way as here ([link](https://github.com/modin-project/modin/pull/3167/files#diff-fd925b89b019dadb68f2bf3d34f86eacebaff1cd85e4256febdac5a7492f2ddaR51))?

Reply:
Discussed offline. Decided to leave as it is.